tests/testthat/test-nrow_subset_linter.R

```diff
@@ -0,0 +1,56 @@
+# TODO(michaelchirico): activate this false positive test when below cases are done.
+# test_that("nrow_subset_linter skips allowed usages", {
+#   # nrow can be avoided here (by chaining the expression and using .N),
+#   #   but the benefit of doing so is not the same as in the other cases.
+#   lintr::expect_lint(
+#     "nrow(DT[x == y, 1, by = grp])",
+#     NULL,
+#     nrow_subset_linter
+#   )
+# })
+
+test_that("nrow_subset_linter blocks subset() cases", {
+  expect_lint(
+    "nrow(subset(x, y == z))",
+    rex::rex("Use arithmetic to count the number of rows satisfying a condition"),
+    nrow_subset_linter()
+  )
+
+  # TODO(michaelchirico): implement this.
+  # lintr::expect_lint(
+  #   "x %>% subset(y == z) %>% nrow()",
+  #   "Use arithmetic to count the number of rows satisfying a condition",
+  #   nrow_subset_linter
+  # )
+})
+
+# TODO(michaelchirico): implement these.
+# test_that("nrow_subset_linter blocks [ cases", {
+#   # data.frame subsetting (NB: replacement doesn't use na.rm = TRUE)
+#   lintr::expect_lint(
+#     "nrow(x[x$y == x$z, ])",
+#     "Use arithmetic to count the number of rows satisfying a condition",
+#     nrow_subset_linter
+#   )
+
+#   # data.table subsetting (NB: replacement needs na.rm = TRUE)
+#   lintr::expect_lint(
+#     "x[y == z, ]",
```

Comment:
Probably you want `.N` in j= here.

Reply:
I think I meant it to be `nrow(x[y == z, ])`... anyway #2315 is the follow-up.